dcm4che-tool/dcm4che-tool-jpg2dcm/src/main/java/org/dcm4che3/tool/jpg2dcm/Jpg2Dcm.java

```diff
@@ -38,377 +38,232 @@
-            newSize <<= 1;
-        }
-        byte[] tmp = new byte[newSize];
-        System.arraycopy(buffer, 0, tmp, 0, jpgHeaderLen);
-        buffer = tmp;
-    }
-
-    private void ensureUID(Attributes attrs, int tag) {
-        if (!attrs.containsValue(tag)) {
-            attrs.setString(tag, VR.UI, UIDUtils.createUID());
-        }
-    }
-
-    private void ensureUS(Attributes attrs, int tag, int val) {
-        if (!attrs.containsValue(tag)) {
-            attrs.setInt(tag, VR.US, val);
+            dos.writeHeader(Tag.SequenceDelimitationItem, null, 0);
+        } finally {
+            if (dos != null)
+                dos.close();
         }
     }
 
     public static void main(String[] args) {
         try {
-            CommandLine cl = parse(args);
+            CommandLine cl = parseCommandLine(args);
             Jpg2Dcm jpg2Dcm = new Jpg2Dcm();
-            if (cl.hasOption(LONG_OPT_CHARSET)) {
-                jpg2Dcm.setCharset(cl.getOptionValue(LONG_OPT_CHARSET));
-            }
-            if (cl.hasOption(LONG_OPT_UID_PREFIX)) {
-                UIDUtils.setRoot(cl.getOptionValue(LONG_OPT_UID_PREFIX));
-            }
-            if (cl.hasOption(LONG_OPT_MPEG)) {
-                jpg2Dcm.setTransferSyntax(UID.MPEG2);
-            }
-            if (cl.hasOption(LONG_OPT_TRANSFER_SYNTAX)) {
-                jpg2Dcm.setTransferSyntax(cl
-                        .getOptionValue(LONG_OPT_TRANSFER_SYNTAX));
-            }
-            jpg2Dcm.setNoAPPn(cl.hasOption(LONG_OPT_NO_APPN));
+            jpg2Dcm.keys = configureKeys(jpg2Dcm, cl);
+            LOG.info("added keys for coercion: \n" + jpg2Dcm.keys.toString());
+            jpg2Dcm.metadataFile = cl.getOptionValue("f");
+            if (cl.getArgs().length < 2)
+                throw new MissingArgumentException("Either input pixel data file or output binary file is missing. See example in jpg2dcm help.");
+            jpg2Dcm.pixelDataFile = new File(cl.getArgs()[0]);
+            Extension ext = getExt(jpg2Dcm.pixelDataFile.getName());
+            jpg2Dcm.noAPPn = Boolean.valueOf(cl.getOptionValue("na"));
+            Attributes metadata = getMetadata(jpg2Dcm, ext);
 
             @SuppressWarnings("rawtypes")
-            List argList = cl.getArgList();
-            File jpgFile = new File((String) argList.get(0));
-            File dcmFile = new File((String) argList.get(1));
+            File dcmFile = new File(cl.getArgs()[1]);
             long start = System.currentTimeMillis();
-            jpg2Dcm.convert(cl, jpgFile, dcmFile);
+            jpg2Dcm.convert(metadata, jpg2Dcm, dcmFile);
             long fin = System.currentTimeMillis();
-            LOG.info("Encapsulated " + jpgFile + " to " + dcmFile + " in "
+            LOG.info("Encapsulated " + jpg2Dcm.pixelDataFile + " to " + dcmFile.getPath() + " in "
                     + (fin - start) + "ms.");
-        } catch (IOException e) {
+        } catch (Exception e) {
```

Comment:
if a ParseException is thrown output the "usage" message as e.g. the storescu tool does:

```
            System.err.println("jpg2dcm: " + e.getMessage());
            System.err.println(rb.getString("usage"));
            System.exit(2);
```